src/LinSys/LinSysMerger.h

```diff
@@ -970,16 +970,14 @@ class LinSysMerger : public CBase_LinSysMerger< HostProxy,
               "Nonzero values of distributed matrix on PE " +
               std::to_string( CkMyPe() ) + " is incomplete: cannot set BCs" );
 
-      for (auto& r : m_lhs) {
-        auto it = m_bc.find( r.first );
-        if (it != end(m_bc)) {
-          auto& diag = tk::ref_find( r.second, r.first );
+      for (const auto& n : m_bc) {
```

Comment:
Documentation for what this group of nested for loops with single character variable names is doing?